File: backend/Models/Product.cs
namespace backend.Models;

public class Product : BaseModel
{
  public string Title { get; set; }
  public string Unit { get; set; }
  public double Price { get; set; }
  public double SalePrice { get; set; }
  public string Description { get; set; }
  public int DiscountInPercent { get; set; }
  public int ProductQuantity { get; set; }
  public string Image { get; set; }
  public Guid CategoryId { get; set; }
  public Category Category { get; set; }
  public IEnumerable<Gallery> Gallery { get; set; }
}

File: theme/packages/admin/src/models/product.ts
import { Category } from "./category"

export interface Product {
  id: string;
  title: string;
  description: string;
  image: string;
  type: string;
  price: number;
  unit: string;
  salePrice: number;
  discountInPercent: number;
  categories: Category[];
  gallery: Gallery[];
}

export type Gallery = {
  url: string;
}

export interface ProductCart extends Product {
  quantity: number;
}
